src/main/java/filter/expression/Qualifier.java

```diff
@@ -93,6 +92,47 @@ public Qualifier(String name, List<SortKey> keys) {
         this.sortKeys = new ArrayList<>(keys);
     }
 
+    public static FilterExpression replaceMilestoneAliases(IModel model, FilterExpression expr) {
+        List<Qualifier> metaQualifiers = expr.find(Qualifier::isMetaQualifier);
+        List<String> repoIds = getReposInMetaQualifiers(metaQualifiers);
+
+        if (repoIds.isEmpty()) {
+            repoIds.add(model.getDefaultRepo().toLowerCase());
+        }
+
+        List<TurboMilestone> allMilestones = model.getMilestones().stream()
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+        Collections.sort(allMilestones, Qualifier::milestoneDueDateComparator);
+
+        int currentMilestoneIndex = getCurrentMilestoneIndex(allMilestones);
```

Comment:
This function should probably return an Optional, for the case where there are no milestones. We should check that before continuing.